### ISSUE TITLE
Inexpressible values after MaxItemsOne flattening under ConfigModeAttr

### ISSUE THREAD

**Issue by VenelinMartinov** [pulumi/pulumi-terraform-bridge#2025]
### What happened?

Combining MaxItemsOne with ConfigModeAttr yields inexpressible values in pulumi land because of MaxItemsOne flattening.

### Example

schema:

```golang
"f0": {
	Optional:   true,
	Type:       schema.TypeList,
	MaxItems:   1,
	ConfigMode: schema.SchemaConfigModeAttr,
	Elem: &schema.Resource{
		Schema: map[string]*schema.Schema{
			"x": {Optional: true, Type: schema.TypeString},
		},
	},
}
```

in terraform we can pass:

```terraform
resource "crossprovider_testres" "example" {
  f0 = []
}
```

but on pulumi side we can either not specify it, pass `x=nil` or `x=non-nil`.

These map to nil, nil, and non-nil respectively but none of them map to [].

### Output of `pulumi about`

.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by t0yv0** [comment]
Yes, somewhat of a known unfortunate issue, thanks for documenting this. This would benefit from fully worked examples and impact.

TF has 5 `unknown`, `nil`, `[]`, [{"foo": "bar"}]` states.

Pulumi has 4 `unknown`, `nil`, `{"foo": "bar"}` states.

We cannot turnaround or map 1:1, but we can decide whether Pulumi nil should map to `nil` or to `[]`, how does it map currently, and what are the trade-offs.
